File: Civitas - Ruby/lib/casilla_impuesto.rb
# To change this license header, choose License Headers in Project Properties.
# To change this template file, choose Tools | Templates
# and open the template in the editor.

module Civitas
  class Casilla_impuesto < Casilla
    def initialize(nombre, importe)
      super
    end
    
    #El jugador paga el impuesto que indique la casilla
    def recibe_jugador(iactual, todos)
      if jugador_correcto(iactual, todos)
        informe(iactual, todos)
        todos[iactual].paga_impuesto(@importe)
      end
    end
    
    def to_s
      return super + "\nCantidad a pagar: " + @importe
    end
  end
end

File: Civitas - Java/src/civitas/CasillaImpuesto.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package civitas;

import java.util.ArrayList;

/**
 *
 * @author juanc
 */
public class CasillaImpuesto extends Casilla{
    
   //Tipo IMPUESTO
    CasillaImpuesto(float cantidad, String nombre){
        super(nombre);
        this.importe = cantidad;
    }
    
    //El jugador paga el impuesto que indique la casilla
    @Override
    protected void recibeJugador(int iactual, ArrayList<Jugador> todos){
        if(jugadorCorrecto(iactual, todos) == true){
            informe(iactual, todos);
            todos.get(iactual).pagaImpuesto(importe);
        }
    }
    
    @Override
    public String toString(){
      return this.nombre + "\nCantidad a pagar: " + this.importe;  
    }
    
}
